packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -190,15 +192,22 @@ export class ConstructNode {
    */
   public findAll(order: ConstructOrder = ConstructOrder.DepthFirst): IConstruct[] {
     const ret = new Array<IConstruct>();
-    const queue: IConstruct[] = [this.host];
+    visit(this.host);
```

Comment:
Why? What was broken about the previous implementation?

Reply:
Changed to pre/post order